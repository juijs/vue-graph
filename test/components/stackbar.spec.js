import { mount } from '@vue/test-utils'
import GraphStackBar from '@/components/stackbar'

describe('stackbar.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphStackBar, {
            props: {
                width: 600,
                height: 400,
                fullMode: false,
                showText: true,
                labels: ['1Q', '2Q', '3Q', '4Q'],
                names: [ 'MS', 'Apple', 'Google' ],
                values: [
                    [ 10, 5, 5, 5 ],
                    [ 40, 10, 10, 10 ],
                    [ 30, 30, 30, 30 ]
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});