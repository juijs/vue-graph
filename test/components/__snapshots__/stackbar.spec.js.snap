// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stackbar.js snapshot 1`] = `
<div
  names="MS,Apple,Google"
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="400"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="600"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="300"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="500"
        x="0"
        y="0"
      />
      <g
        class="grid-block"
        transform="translate(50,350)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            1Q
          </text>
        </g>
        <g
          transform="translate(125,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            2Q
          </text>
        </g>
        <g
          transform="translate(250,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            3Q
          </text>
        </g>
        <g
          transform="translate(375,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 62.5,13)"
            x="62.5"
            y="13"
          >
            4Q
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="500"
          y1="0"
          y2="0"
        />
        <g
          transform="translate(500,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
        </g>
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-range"
        transform="translate(50,50)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            80
          </text>
        </g>
        <g
          transform="translate(0,30)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            72
          </text>
        </g>
        <g
          transform="translate(0,60)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            64
          </text>
        </g>
        <g
          transform="translate(0,90)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            56
          </text>
        </g>
        <g
          transform="translate(0,120)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            48
          </text>
        </g>
        <g
          transform="translate(0,150)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            40
          </text>
        </g>
        <g
          transform="translate(0,180)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            32
          </text>
        </g>
        <g
          transform="translate(0,210)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            24
          </text>
        </g>
        <g
          transform="translate(0,240)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            16
          </text>
        </g>
        <g
          transform="translate(0,270)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            8
          </text>
        </g>
        <g
          transform="translate(0,300)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            0
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="0"
          y1="0"
          y2="300"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="300"
          stroke="transparent"
          width="500"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-stackcolumn"
        transform="translate(50,50)"
      >
        <g>
          <rect
            fill="#7977C2"
            height="37.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="2"
            y="262.5"
          />
          <rect
            fill="#7BBAE7"
            height="150"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="2"
            y="112.5"
          />
          <rect
            fill="#FFC000"
            height="112.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="2"
            y="0"
          />
          <text
            dx="0"
            dy="-3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="bold"
            opacity="0"
            text-anchor="middle"
            transform="translate(62.5,0)"
          >
            80
          </text>
        </g>
        <g>
          <rect
            fill="#7977C2"
            height="18.75"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="127"
            y="281.25"
          />
          <rect
            fill="#7BBAE7"
            height="37.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="127"
            y="243.75"
          />
          <rect
            fill="#FFC000"
            height="112.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="127"
            y="131.25"
          />
          <text
            dx="0"
            dy="-3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="bold"
            opacity="0"
            text-anchor="middle"
            transform="translate(187.5,131.25)"
          >
            45
          </text>
        </g>
        <g>
          <rect
            fill="#7977C2"
            height="18.75"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="252"
            y="281.25"
          />
          <rect
            fill="#7BBAE7"
            height="37.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="252"
            y="243.75"
          />
          <rect
            fill="#FFC000"
            height="112.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="252"
            y="131.25"
          />
          <text
            dx="0"
            dy="-3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="bold"
            opacity="0"
            text-anchor="middle"
            transform="translate(312.5,131.25)"
          >
            45
          </text>
        </g>
        <g>
          <rect
            fill="#7977C2"
            height="18.75"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="377"
            y="281.25"
          />
          <rect
            fill="#7BBAE7"
            height="37.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="377"
            y="243.75"
          />
          <rect
            fill="#FFC000"
            height="112.5"
            stroke="none"
            stroke-opacity="0"
            stroke-width="0"
            width="121"
            x="377"
            y="131.25"
          />
          <text
            dx="0"
            dy="-3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="bold"
            opacity="0"
            text-anchor="middle"
            transform="translate(437.5,131.25)"
          >
            45
          </text>
        </g>
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
